### ISSUE TITLE
Document our OMI Github Process

### ISSUE THREAD

**Issue by mrmetaverse** [omigroup/website#34]
Create an easy to follow how-to document for how we use github to manage our work at OMI.

**Comment by mrmetaverse** [comment]
Started draft here on [hackmd](https://hackmd.io/@mrmetaverse/omi-github-process)

**Comment by mrmetaverse** [comment]
For review [on hackmd](https://hackmd.io/@mrmetaverse/omi-github-process)


# OMI Github Strategy

This plan highlights how we can use github to handle the majority of our working processes. Github allows us to reduce the number of tools we are using, and keep everything in one place.

It can be challenging for newcomers to get involved, we want to ensure that everyone understands where work is happening, and establish a reliable place for future work to live. 


## Github Discussions:

Sometimes we need to talk something through together. When you need group insight and collaboration, slack can get quick answers but often key insights are lost to day to day communications and noise. Documenting our process doesn't have to be a burden put off until after you are finished. Documentation can be captured, as you go, by adopting a few simple processes. 

![](https://hackmd.io/_uploads/S14b_Zq4F.png)

### Uses: 

- When you don't have a specific problem or solution identified and need to talk it out or brainstorm ideas
- Create one as meeting notes at the start of any meeting and collaboratively comment observations and links and demos in the comments.
    - :star: Even better: Create a discussion with just the agenda for an upcoming meeting. Then take notes together in the comments on meeting day. 
- When you want to reflect on a workshop
- When a official artifact is in draft mode
>note: you can convert comments from discussions into backlog issues. A discussion can quickly create a list of backlog items _while you brainstorm_.

## Backlogs / Kanban boards / Github "Projects"

At times, we must juggle a lot of things at once :juggling: and this can cause confusion, spinning tires, and analysis paralysis. Just as someone gets up to speed, a new person joins and it starts over again. 

Use the backlog(s) which are called _"Projects"_ on github. 

![](https://hackmd.io/_uploads/Sy2HOWqEY.png)

>note: There are now github projects, AND "projects-_beta_" boards. I recommend using the standard github projects boards for now, as the _beta_ boards are brand new. 

### Purpose of the backlog: 
We need a place to reliably put our **known problems to solve,** prioritize work, and track progress. An item is moved from the to-do column, to the in progress column when work for that item starts. If the ticket becomes blocked, or is ready for review, the card moves to that column. And upon approval, or termination, the issue will move to the "done" column. 

This allows us to track any amount of work, and flesh out our ideas as we go. 

![](https://hackmd.io/_uploads/r1kTdW94t.png)

### We have multiple Projects / Backlogs 

OMI is a fluid group consisting of a number of working groups that self-organize around common goals. These backlogs help organize this work at the working-group level, and we can syndicate all active issues on the [OMI Holisitc Backlog](https://github.com/orgs/omigroup/projects/2) to see all in-flight issues in one place. 

If someone couldn't make it to a meeting this week, having a place to reliably see what work is in flight, who is doing what, and how to get involved is critical to a self-governing and organic group like OMI. 

Think of repos like sandboxes for themes of work or working groupos. So far we have backlogs in `github.com/omigroup/omi` for managing general tasks, `github.com/omigroup/website` for tracking website tasks and issues, `github.com/omigroup/gltf-extensions` for managing the workflows of the gltf extensions group, and more. 

You can actually create a new project by clicking `New Project` in the "Projects" menu of any repo. 


### Holistic Overview: 
This is a holistic backlog for general operations and tracking key tasks across all our working groups repos and projects. This backlog lives at the Org level at `github.com/omigroup/` and can be accessed by clicking Projects in on the omigroup organization page. 

![](https://hackmd.io/_uploads/SycfkMcNK.png)
![](https://hackmd.io/_uploads/B1Tu1G5EF.png)


### How to use a Github Projects backlog 
>#### To add a card click the `+` symbol at the top right of the column 

### Add a Note 

Notes on the backlog can be easily identified with white text. 

![](https://hackmd.io/_uploads/BkbU5b54F.png)

There is no clickable link, just text and markdown. You can add tags to a note, add markdown, and quick reminders. Notes are like quick reminders. Something like "remind team to submit daily balance scores" or "schedule first interview". 


>:rotating_light: If you do not have access to add tasks or issues to the backlog, and feel you need too, please reach out to mrmetaverse for github permissions.

### Create an issue 

Issues are a larger chunk of work, that you may need to discuss, track, or breakdown. Issues can be easily identified by having blue text which can be clicked. 

![](https://hackmd.io/_uploads/Hkjvcb5VK.png)

Maybe you already know the work that needs to be done, so you may write an issue like you would write a user story, with acceptance criteria and  definition of done. 

In other cases, you may have to convert a comment on a discussion, or a note on a backlog convert it into an issue. Issues allow you to assign tags, add owners, add additional details, discuss in comments, and more. 

Issues are often used to call out feature requests, bug reports, and requests for help. Issues are fodder for pull requests and recommendations. 

At any time you can click on `Go to issue for full details` to open up that issues page, and see comments, changes, files, tags, and other information: 

![](https://hackmd.io/_uploads/ByNa9b9EY.png)


#### Adding Tags / Labels: 

You can tag issues based on which objective or core value they are a part of, like `documentation` or `Maintain Sustainable Innovation` and you can also add tags to fit your needs. 

![](https://hackmd.io/_uploads/HyWNibqNY.png)

To add a Tag, click the blue text of a card of interest. A tray will slide out on the right side.

![](https://hackmd.io/_uploads/B1xIjbq4K.png)


You may then assign tags by searching and clicking on the ones that are relevant. 

![](https://hackmd.io/_uploads/rkEKibqNK.png)


#### Adding issues to multiple projects/backlogs: 

![](https://hackmd.io/_uploads/BkG3jb5VF.png)


If you are working on a specific working group like `omigroup/website`, you will be working from the [OMI Website Groups Backlog](https://github.com/omigroup/website/projects/1). However, when you add a new item of interest, it is helpful to add that issue to the [OMI Holisic backlog](https://github.com/orgs/omigroup/projects/2) as well. This helps track all of the work in flight across both projects. 

#### Converting a discussion comment into an issue

Discussions are a great way to generate multiple responses. Maybe you create a discussion that asks "How can we identify which problems matter most to the client?" The team can respond in comments with suggestions like, "what if we call them?" or "How about a LDJ workshop?" 

![](https://hackmd.io/_uploads/H1ms2ZcEF.png)

You can convert that comment into a new issue on the backlog: 

![](https://hackmd.io/_uploads/ry83hZqEK.png)

Just select `Reference in new issue` and add some context for your new issue. 

![](https://hackmd.io/_uploads/HJLRh-9Et.png)


#### Converting a backlog note into an issue 

On our projects boards / backlogs, we can quickly add notes for reminders. But maybe that escilated in complexity. Maybe you had a note to `remind the team of new meeting time Friday at 6pm`. You go to complete that task, and the team responds negatively. "What do you mean we should meet on Fridays at 6? I think we should talk about this". 

Well you can't have a discussion within a note, only issues and discussions. Since this has turned into a bigger topic, you can get a lot of responses async in comments of an issue. To convert a note to an issue, click the `...` at the top of the note: 

![](https://hackmd.io/_uploads/ryFfpZ5NK.png)

and select `Convert to issue` and add a description. 
![](https://hackmd.io/_uploads/rktraW5EY.png)


## the README 

The [omigroup/OMI readme](https://github.com/omigroup/OMI) is a source of truth. Often, a readme is used to explain how to pull code, run and test code, and submit changes.

In our research efforts, a readme is a great place to summarize our work. Who are we serving, Why are we here, What are our objectives, and How do we plan to measure progress. One brief portal that helps newcomers get up to speed, and understand where and how to get involved. 

![](https://hackmd.io/_uploads/HkkDlM5VF.png)


## Adding files 

You may upload files directly into the repo. First, select which folder you want to add files to: 

![](https://hackmd.io/_uploads/rydWMz9EF.png)

You will be brought to a new page. You can upload directly here, or to the artifacts folder by click `assets`. 

Once you are where you want to add your file, just click `Add file` next to the "Go to file" button.  

![](https://hackmd.io/_uploads/Hk77fz9Et.png)

You may then add a file name, and description. This is a change to add links, or additional comments as well. 

:rotating_light: Avoid committing directly to master. Instead select "create a new branch for this commit and start a pull request. Learn more about pull requests." and click `propose changes` when finished

![](https://hackmd.io/_uploads/SyT5MfcNt.png)


## Creating a new repo 

Developers may need to establish a new repo entirely. Maybe the project turns into 3-5 other product lines or solutions. In this case, you can still cross post issues on multiple backlogs, and link to these repos in readmes and documentation. 

## The wiki 

Once a discussuon is complete, or a issue has been closed, a final artifact may have been created. We can save these artifacts, and key insights in our `omigroup/omi` Wiki. Like a final draft for future readers to get up to speed, and for rapid reference on the fly. 

- User groups 
- Key problems to solve 
- Product vision 
- Working agreement 

>At present, we do not use the github wiki feature at OMI

## What if we hate it? 

Use it! Have conversations, reflect on if its working, lay breadcrumbs for future team members. But above all, we must try. Now at this point, I challenge the team to add your unknowns to the backlogs, and ensure that you have all you need defined.

**Comment by mrmetaverse** [comment]
Upon acceptance, I propose moving this to our omigroup/omi wiki. I also suggest adding "Our Vision," "Our Core Values," "Our Backstory," "How to submit a request for review to OMI," etc.

**Comment by mrmetaverse** [comment]
Added a section for using our issue templates
